### PR TITLE
support json response for wxa login

### PR DESCRIPTION
**Pull request by XeniaLu**

### CODE REVIEW

File: src/gapp_login/wechat/endpoints.py [hunk 1]
```diff
@@ -80,6 +80,10 @@ async def login_wechat(
         rv = await auth.create_authorization_response(request, user, ctx)
         if rv.status_code >= 400:
             tx.raise_rollback()
+        rv_location = rv.headers["location"]
+        if rv_location.startswith("wxa://"):
+            params = parse.parse_qs(rv_location.split("?", 1)[1])
+            return auth.handle_response(200, dict(code=params.get("code")), {})
```
Comment (fantix) [inline src/gapp_login/wechat/endpoints.py]: should be fine to return just a `dict` innstance

Comment (fantix) [inline src/gapp_login/wechat/endpoints.py]: other than `code`, I think `state` is also needed

**Review by fantix — CHANGES_REQUESTED** [review]

File: src/gapp_login/wechat/endpoints.py (hunk 1 above)
Comment (XeniaLu) [reply src/gapp_login/wechat/endpoints.py]: right, fixed~

Comment (XeniaLu) [reply src/gapp_login/wechat/endpoints.py]: right, fixed

Comment (XeniaLu) [reply src/gapp_login/wechat/endpoints.py]: fixed

**Review by fantix — APPROVED** [review]

**Pull request merged**